### ISSUE TITLE
Longevity test 2023-12-20 OOM

### ISSUE THREAD

**Issue by lmatz** [risingwavelabs/risingwave#14078]
### Describe the bug

run all nexmark (8 sets of nexmark queries) with 10k throughput

RW Version:
nightly-20231219  

Grafana Metric:
https://grafana.test.risingwave-cloud.xyz/d/EpkBw5W4k/risingwave-dev-dashboard?orgId=1&var-datasource=Prometheus:%20test-useast1-eks-a&var-namespace=reglngvty-20231219-150239&from=1702998973000&to=1703042305000

Grafana Logs:
https://grafana.test.risingwave-cloud.xyz/d/liz0yRCZz1/log-search-dashboard?orgId=1&var-data_source=Logging:%20test-useast1-eks-a&var-namespace=reglngvty-20231219-150239&from=1702998973000&to=1703042305000

Buildkite Job:https://buildkite.com/risingwave-test/longevity-test/builds/865

```
benchmark-risingwave-compute-c-1
Last State:     Terminated
--
  | Reason:       OOMKilled
  | Exit Code:    137
  | Started:      Tue, 19 Dec 2023 15:06:20 +0000
  | Finished:     Wed, 20 Dec 2023 01:18:51 +0000
```

Will decode the memory profile dump, TBC......
Link: https://s3.console.aws.amazon.com/s3/buckets/test-useast1-mgmt-bucket-archiver?prefix=k8s/reglngvty-20231219-150239/&region=us-east-1

**Comment by lmatz** [comment]
There are too many dumps, so I just randomly picked a few
[dumps.zip](https://github.com/risingwavelabs/risingwave/files/13791711/dumps.zip)

![SCR-20231229-ls6](https://github.com/risingwavelabs/risingwave/assets/5791930/3fec6f95-adb0-492c-81e0-b277d7c6cb16)

![SCR-20231229-lty](https://github.com/risingwavelabs/risingwave/assets/5791930/9b165d97-3e52-48c3-a121-1ff3eb344f27)

**Comment by xxchan** [comment]
As mentioned https://github.com/risingwavelabs/risingwave/issues/14324#issuecomment-1875145121, this one is our old behavior, and not fixed by #14333. The kafka splits are randomly assigned to 3 worker node's actors, and it could be uneven.
